### ISSUE TITLE
Unable to determine item related to a Job Card in multi-level BOM work order.

### ISSUE THREAD

**Issue by tlundy-optimation** [frappe/erpnext#17872]
Steps to reproduce:

1.  Create Part 1 and Part 2 with at least one identical operation.  In my case, both have a nesting and cutting operation.
2.  Create a sub assembly of 2 parts (1 each).  I call this Sub 1 consisting of Part 1 and Part 2 welded together.
3.  Create a work order for Sub 1 using multi level BOM.   Mine was an order for 10.
4.  After submitting it you will have 5 job cards created.   2 nesting jobs, 2 cutting jobs and 1 welding job.
5.  You will not be able to tell by looking at the job card information in the UI or in the Rest API, which is for Part 1 and which for Part 2.  The parent work order only refers to Sub 1 and the BOM entry is the BOM for Sub 1.  

This seems like a general error, since it would be impossible to report times back to a Job Card because you wouldn't know which part the card was associated with.  In my specific case, I am integrating this with our nesting system that pulls orders for a machine and reports back units completed, times and material usage for both the nesting and cutting operations.   I cannot even pull the orders for a specific operation because I have no way of finding out which item the Job Card for 10 nested parts is referring to.  

My suggested fix would be to refer to the item name in every job card, but I am open to other methods.  This is not an issue with Job Cards using a single level BOM because the work order has the item name stored against it and I can look it up.  However, even in this case it would be more convenient if I didn't have to make a separate set of calls to retrieve the work orders for each job card.  Examples of the Rest responses and UI screens for Part 1 and Part 2 job cards are attached for the above case.

Thanks,
Tate

Example JSON response from the Rest API for the two nesting job cards for different parts on the same multi-level BOM order:

{
“data”: {
“for_quantity”: 10.0,
“total_completed_qty”: 0.0,
“doctype”: “Job Card”,
“owner”: ,
“operation”: “Nest”,
“work_order”: “MFG-WO-2019-00003”,
“modified_by”: ,
“time_logs”: [],
“transferred_qty”: 0.0,
“docstatus”: 0,
“requested_qty”: 0.0,
“status”: “Open”,
“creation”: “2019-05-28 16:54:54.482143”,
“job_started”: 0,
“company”: ,
“workstation”: “Laser Nesting”,
“posting_date”: “2019-05-28”,
“wip_warehouse”: “Work In Progress - O”,
“bom_no”: “BOM-Sub 1-001”,
“name”: “PO-JOB00011”,
“idx”: 0,
“actual_material_usage”: 0.0,
“modified”: “2019-05-28 16:54:54.482143”,
“items”: [],
“operation_id”: “22d25c5274”,
“total_time_in_mins”: 0.0
}
}

{
“data”: {
“for_quantity”: 10.0,
“total_completed_qty”: 0.0,
“doctype”: “Job Card”,
“owner”: ,
“operation”: “Nest”,
“work_order”: “MFG-WO-2019-00003”,
“modified_by”: ,
“time_logs”: [],
“transferred_qty”: 0.0,
“docstatus”: 0,
“requested_qty”: 0.0,
“status”: “Open”,
“creation”: “2019-05-28 16:54:54.506470”,
“job_started”: 0,
“company”: ,
“workstation”: “Laser Nesting”,
“posting_date”: “2019-05-28”,
“wip_warehouse”: “Work In Progress - O”,
“bom_no”: “BOM-Sub 1-001”,
“name”: “PO-JOB00012”,
“idx”: 0,
“actual_material_usage”: 0.0,
“modified”: “2019-05-28 16:54:54.506470”,
“items”: [],
“operation_id”: “8c499649bc”,
“total_time_in_mins”: 0.0
}
}

![image](https://user-images.githubusercontent.com/26491515/59042571-4b0df080-8840-11e9-8055-4f08fa4742b9.png)
![image](https://user-images.githubusercontent.com/26491515/59042708-90cab900-8840-11e9-9bcc-c53aefeb3dc4.png)

**Comment by govindsmenokee** [comment]
I am facing the same issue. Job Card is having the same BOM for main assembly and sub assembly

**Comment by nameduser0** [comment]
Yes, this seems crazy but I guess it was first implemented for single level BOMs and wasn't extended properly to collate several BOMs together in the work order. The operation absolutely needs the part in question bringing through when compiling the work order from the BOMs.

The only workaround seems to be to have different operation names for individual parts!

Can't believe this hasn't been picked up in 2 years!

**Comment by sharmanakul** [comment]
@ankush https://frappe.io/app/backlog/BCK-LG-2021-0612
